tests/pamiq_core/threads/test_thread_control.py

```diff
@@ -257,3 +261,203 @@ def inifinity_count():
+        thread_status.resume()
+        threading.Timer(0.1, thread_status.pause).start()
+        start = time.perf_counter()
+        assert thread_status.wait_for_pause(0.5) is True
+        assert 0.1 <= time.perf_counter() - start < 0.2
+
+
+class TestReadOnlyThreadStatus:
+    def test_exposed_methods(self):
+        """Test of exposure of functions from ThreadStatus."""
+        thread_status = ThreadStatus()
+        read_only_thread_status = ReadOnlyThreadStatus(thread_status)
+        assert read_only_thread_status.is_pause == thread_status.is_pause
+        assert read_only_thread_status.is_resume == thread_status.is_resume
+        assert read_only_thread_status.wait_for_pause == thread_status.wait_for_pause
+
+
+class TestThreadStatusesHandler:
+    @pytest.fixture()
+    def inference_thread_status(self):
+        """Fixture for thread status, used for inference."""
+        return ThreadStatus()
+
+    @pytest.fixture()
+    def read_only_inference_thread_status(self, inference_thread_status):
+        """Fixture for read-only thread status, used for inference."""
+        return ReadOnlyThreadStatus(inference_thread_status)
+
+    @pytest.fixture()
+    def training_thread_status(self):
+        """Fixture for thread status, used for training."""
+        return ThreadStatus()
+
+    @pytest.fixture()
+    def read_only_training_thread_status(self, training_thread_status):
+        """Fixture for read-only thread status, used for training."""
+        return ReadOnlyThreadStatus(training_thread_status)
+
+    @pytest.fixture()
+    def thread_status_handler(
+        self, read_only_inference_thread_status, read_only_training_thread_status
+    ):
+        """Fixture for thread status handler."""
+        return ThreadStatusesHandler(
+            {
+                ThreadTypes.INFERENCE: read_only_inference_thread_status,
+                ThreadTypes.TRAINING: read_only_training_thread_status,
+            }
+        )
+
+    def test_wait_for_all_threads_pause_when_empty_status(self):
+        """Test wait_for_all_threads_pause when statuses is empty."""
+        # immediately return True if statuses is empty
+        thread_status_handler = ThreadStatusesHandler(statuses={})
+        start = time.perf_counter()
+        assert thread_status_handler.wait_for_all_threads_pause(0.1) is True
+        assert time.perf_counter() - start < 1e-3
+
+    def test_wait_for_all_threads_pause_all_when_all_threads_paused(
+        self, inference_thread_status, training_thread_status, thread_status_handler
+    ):
+        """Test wait_for_all_threads_pause when all threads are paused."""
+        # immediately return True if all threads are paused
+        inference_thread_status.pause()
+        training_thread_status.pause()
+
+        start = time.perf_counter()
+        assert thread_status_handler.wait_for_all_threads_pause(0.1) is True
+        assert time.perf_counter() - start < 2e-3  # test not passed if 1e-3
```

Comment:
ThreadPoolExecutorの立ち上げが割と処理がかかるんですね。 もっと余裕を持っても良いかもしれません。
```suggestion
        assert time.perf_counter() - start < 1e-2  # test not passed if 1e-3
```

Reply:
この方向で修正しました！